services/static-webserver/client/source/class/osparc/data/model/Node.js

```diff
@@ -1327,68 +1328,51 @@ qx.Class.define("osparc.data.model.Node", {
       }
     },
 
-    __waitForServiceReady: async function(srvUrl) {
+    __waitForServiceReady: function(srvUrl) {
       this.getStatus().setInteractive("connecting");
+
+      if (this.__retries === 0) {
+        return;
+      }
+
       const retry = () => {
+        this.__retries--;
+
         // Check if node is still there
         if (this.getWorkbench().getNode(this.getNodeId()) === null) {
           return;
         }
-        const interval = 3000;
+        const interval = 5000;
         qx.event.Timer.once(() => this.__waitForServiceReady(srvUrl), this, interval);
       };
 
       // ping for some time until it is really reachable
       try {
         if (osparc.utils.Utils.isDevelopmentPlatform()) {
-          console.log("Connecting: about to fetch ", srvUrl);
-        }
-        const response = await fetch(srvUrl);
-        if (osparc.utils.Utils.isDevelopmentPlatform()) {
-          console.log("Connecting: fetch's response ", JSON.stringify(response));
+          console.log("Connecting: about to fetch", srvUrl);
         }
-        if (response.ok || response.status === 302) {
-          // ok = status in the range 200-299
-          // some services might respond with a 302 which is also fine
-          // instead of
-          // - requesting its frontend to make sure it is ready and ...
-          // - waiting for the "load" event triggered by the content of the iframe
-          // we will skip those steps and directly switch its iframe
-          this.__serviceReadyIn(srvUrl);
-        } else {
-          console.log(`Connecting: ${srvUrl} is not reachable. Status: ${response.status}`);
-          retry();
-        }
-      } catch (error) {
-        console.error(`Connecting: Error while checking ${srvUrl}:`, error);
-        retry();
-      }
-    },
-
-    __waitForServiceWebsite: function(srvUrl) {
-      // request the frontend to make sure it is ready
-      let retries = 5
-      const openAndSend = () => {
-        if (retries === 0) {
-          return
-        }
-        retries--
         fetch(srvUrl)
-          .then(request => {
-            if (request.status >= 200 || request.status < 300) {
-              this.__serviceReadyIn(srvUrl)
+          .then(response => {
+            if (osparc.utils.Utils.isDevelopmentPlatform()) {
+              console.log("Connecting: fetch's response status", response.status);
+            }
+            if (response.ok || response.status === 302) {
+              // ok = status in the range 200-299
+              // some services might respond with a 302 which is also fine
```

Comment:
why not just `response.status<400` ?? This is normally considered an `ok` response
- https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.ok
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
IMO what you are missing here is "who" is reponding ... 
The iFrame should render whatever the "final" backends service gives you. It could even be a 5XX ... does not matter. You should just wait until you know that the response is not coming from any of the intermediate services.
Typically that information is found in the response headers.  We could add some custom header to identify the intermediaries (that we control) so you can discriminate when is **not** coming from those.

Reply:
On our side an ok is a 200-299:
https://developer.mozilla.org/en-US/docs/Web/API/Response/ok

Reply:
@pcrespov I disagree that the frontend should know anything about who responds from inside. We should not introduce such dependencies.
I agree that it should retry whatever the response is (4xx <= x < 6xx).